### PR TITLE
Guide: Role assignment on reaction

### PR DESCRIPTION
**Pull request by queezly**
A common asked question has been "How do i add a role to someone that reacts to the bots message"
This PR adds a guide and code sample on the topic categorized under Popular Topics > Reactions

### CODE REVIEW

**Comment by Danktuary** [comment]
@Drahcirius brought up a good point about 2 weeks ago.
>I'm not sure about using a collector for role react though
>if they're going to use a command, why wouldn't the command just give them the role directly

>I can already see people making time 99999999999
>and having tons of collectors
>and every time their bot restarts it dies

It didn't hit me until that 2nd part, but some people might misunderstand how this works exactly. Coincidentally, someone in the support channels did that exact thing a few days after Drah mentioned this. Most people are looking for a more permanent solution, which would pretty much require using the regular `messageReactionAdd` event and `raw` event as well.

As for the PR, I'm not sure if you wanna rework the example or just leave it alone overall. I'd suggest a different example but I honestly can't think of one. Alternatively, if you're up for it and want to create a full page for that kind of stuff, that'd be welcome. If you're not, don't worry, I wouldn't blame you.

File: guide/popular-topics/reactions.md [hunk 1]
```diff
@@ -176,6 +176,33 @@ message.awaitReactions(filter, { max: 1, time: 60000, errors: ['time'] })
...
+await reactmessage.react('👌'); 
+
+const filter = (reaction, user) => reaction.emoji.name === '👌' && !user.bot; 
+const collector = reactmessage.createReactionCollector(filter, { time: 15000 });
+
+collector.on('collect', async reaction => { 
+	const user = reaction.users.last(); // the last user that reacted aka the user that triggered the 'collect' event
+	const guild = reaction.message.guild; // the guild the reaction happened in
+	const member = guild.member(user) || await guild.fetchMember(user); // getting the member from the user
+	member.addRole("some-role-id"); // adding the role to the member
```
Comment (xDimGG) [inline guide/popular-topics/reactions.md]: Quote consistency

File: guide/popular-topics/reactions.md [hunk 2]
```diff
@@ -176,6 +176,33 @@ message.awaitReactions(filter, { max: 1, time: 60000, errors: ['time'] })
...
+	console.log(`Added the role to ${member.displayName}`);
+});
+```
+<p class="tip">On the masterbranch the `collect` event emits with the User that reacted as second parameter, so you don't need to obtain it as `reaction.users.last()`</p>
```
Comment (xDimGG) [inline guide/popular-topics/reactions.md]: "masterbranch"

Comment (xDimGG) [inline guide/popular-topics/reactions.md]: No reason to capitalize "user" here

File: guide/popular-topics/reactions.md [hunk 3]
```diff
@@ -176,6 +176,33 @@ message.awaitReactions(filter, { max: 1, time: 60000, errors: ['time'] })
...
+
+If you want a static massage on the server people can react to to assign their roles you are better off using the `messageReactionAdd` event and comparing the `reaction.message.id` to your specified reactionmessage. This clientevent passes you the [MessageReaction object](https://discord.js.org/#/docs/main/stable/class/MessageReaction) and the user that reacted which you can utilize in the same way as shown above.
```
Comment (xDimGG) [inline guide/popular-topics/reactions.md]: "massage"

Comment (xDimGG) [inline guide/popular-topics/reactions.md]: "reactionmessage" "clientevent"

**Comment by queezly** [comment]
Will close this, as the proposed idea needs a lot more rework and planning

**Pull request closed without merging**